### PR TITLE
Enable FileMutex on Window

### PR DESCRIPTION
**Pull request by davidsonalencar**
| Q | A |
| --- | --- |
| Is bugfix? | yes |
| New feature? | no |
| Breaks BC? | no |
| Tests pass? | no |
| Fixed issues | comma-separated list of tickets # fixed by the PR, if any |

Suggestion for enable Mutex for Windows. By my tests with this change this feature works on Windows.

### CODE REVIEW

**Comment by samdark** [comment]
Are you sure it works correctly as a mutex?

**Comment by samdark** [comment]
This is why it's disabled on Windows:

```php
flock($file, LOCK_EX | LOCK_NB)
```

**Comment by davidsonalencar** [comment]
Based on the [PHP documentation](http://php.net/manual/en/function.flock.php), the function `flock` works on Windows with limitation, but meets to the requirement of the `FileMutex`.

**Comment by davidsonalencar** [comment]
I don't see why close this issue...

**Comment by samdark** [comment]
@yiisoft/core-developers do you remember why exactly it's disabled on Windows?

**Comment by klimov-paul** [comment]
`FileMutex` was created at #580 by @resurtm . May be he remebers.

**Comment by klimov-paul** [comment]
> Note: flock() uses mandatory locking instead of advisory locking on Windows.

**Comment by kidol** [comment]
When code was created in 2013, PHP docs were wrong on this.

https://bugs.php.net/bug.php?id=65272
>2015-02-02 18:46 UTC
>The docs are updated to reflect the fact that LOCK_NB does indeed work on Windows systems.

File: framework/mutex/FileMutex.php [hunk 1]
```diff
@@ -137,6 +134,6 @@ protected function releaseLock($name)
      */
     protected function getLockFilePath($name)
     {
-        return $this->mutexPath . '/' . md5($name) . '.lock';
+        return $this->mutexPath . DIRECTORY_SEPARATOR . md5($name) . '.lock';
```
Comment (samdark) [inline framework/mutex/FileMutex.php]: No need to change that. It works well as is on Windows.

**Review by samdark — CHANGES_REQUESTED** [review]
Need a changelog.

File: framework/mutex/FileMutex.php (hunk 1 above)
Comment (dynasource) [inline framework/mutex/FileMutex.php]: Im considering '/' to be a best practice and to use DIRECTORY_SEPARATOR only when there is some matching to do.

**Comment by davidsonalencar** [comment]
Added CHANGELOG change in the pull request,

**Comment by samdark** [comment]
Thank you!

**Comment by davidsonalencar** [comment]
I thank you!

**Pull request merged**